File: modules/chat-core/src/main/java/com/shterneregen/securelan/chat/service/ChatServerConfig.java
package com.shterneregen.securelan.chat.service;

import java.util.Objects;

public record ChatServerConfig(int port, String sessionPassword) {
    public ChatServerConfig {
        Objects.requireNonNull(sessionPassword, "sessionPassword must not be null");
        if (port < 1 || port > 65535) {
            throw new IllegalArgumentException("port must be between 1 and 65535");
        }
    }
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/service/ChatServerConfig.kt
package com.shterneregen.securelan.chat.service

import java.util.Objects

@JvmRecord
data class ChatServerConfig(val port: Int, val sessionPassword: String) {
    init {
        Objects.requireNonNull(sessionPassword, "sessionPassword must not be null")
        require(port in 1..65_535) { "port must be between 1 and 65535" }
    }
}
